lib/kernel/isinf_half.cl

```diff
@@ -0,0 +1,34 @@
+/* OpenCL built-in library: isinf()
+
+   Copyright (c) 2024 Henry Linjamäki / Intel Finland Oy
+
+   Permission is hereby granted, free of charge, to any person obtaining a copy
+   of this software and associated documentation files (the "Software"), to deal
+   in the Software without restriction, including without limitation the rights
+   to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+   copies of the Software, and to permit persons to whom the Software is
+   furnished to do so, subject to the following conditions:
+
+   The above copyright notice and this permission notice shall be included in
+   all copies or substantial portions of the Software.
+
+   THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+   IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+   FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+   AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+   LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+   OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+   THE SOFTWARE.
+*/
+
+#include "templates.h"
+
+int __attribute__((overloadable)) isinf(half a) {
+  return __builtin_isinf(a);
+}
+
+IMPLEMENT_BUILTIN_L_V(isinf, short2, half2, half, lo, hi)
```

Comment:
Is it necessary to add a new file only for halfs ? The existing macros in `lib/kernel/isinf.cl` should already automatically define the half variants of isinf when it's available, so i think we should just use that file (and fix it if necessary).

Reply:
Question is that is the libclc's isinf() implementation preferred over `lib/kernel/isinf.cl` or could/should the libclc's implementation be discarded?

Reply:
What is the upstream? The libclc files hosted in PoCL doesn't look like they are from llvm-project/libclc at all.

Reply:
> look like they are from llvm-project/libclc

They are from libclc (sort of). IIRC files in `lib/kernel/libclc-pocl` are wrappers (for compiling the same sources with different vector width) and the sources in `lib/kernel/libclc` are originally from libclc adapted to PoCL. They are not upstreamable, PoCL is the upstream.

> Question is that is the libclc's isinf() implementation preferred

No, it's not preferred. The PoCL's libclc sources do not have half support, because IIRC half support was almost nonexistent in both hardware and software at the time this was adapted from libclc (2017 i think).

Reply:
OK, maybe it's so old fork that it has diverged heavily upstream.